File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private Queue q = new LinkedList();
    private List<Integer> result = new ArrayList();
    
    private void dfs(TreeNode node) {
        if (node != null) {
            dfs(node.left);
            result.add(node.val);
            dfs(node.right);
            return;
        }
        return;
    }

    public List<Integer> inorderTraversal(TreeNode root) {
        if (root == null) return result;
        dfs(root);
        return result;
    }
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function inorderTraversal(root: TreeNode | null): number[] {
    if (!root) return [];
    const result = [];

    function dfs(node: TreeNode | null) {
        if (node) {
            if (node.left) dfs(node.left);
            if (node.val !== null) result.push(node.val);
            if (node.right) dfs(node.right);   
        }
    }
    dfs(root);
    return result;
};